### PR TITLE
feat(couch setup):#8-Couch base api added

### PR DESCRIPTION
**Pull request by karthikeyanpa12**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/web-slate/simple-google-search/blob/main/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What type of change does this PR introduce?

- [x] API
- [ ] Frontend

## PR Sub Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #8

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Step. 1:
     **Couch base connection details:**
  
          couchbase.connection-string=127.0.0.1
          couchbase.bucket-name=google-search
          couchbase.user-name=user_profile
          couchbase.password=Administrator

Step 2:
     Create Primary Index in couch base to get the details
   
    Query:
        create primary index `primary_google_search_index` on `google-search` using GSI;

      * primary_google_search_index =  index name
      * google-search = bucket name

Step 3 : 
       Created below files for addSearchResult and getSearchResult API

         Model = Search.java
         Repository = SearchRepository.java for CRUD operations

Step. 4:
    API Details

      1) AddSearch - http://localhost:8080/api/v1/addSearch

           Request:
                 { "keyword": "facebook", "result": "facebook,instagram,whatsapp"}

    2) getSearch - http://localhost:8080/api/v1/getSearch

           Response:
                 [{"keyword": "facebook", "result": "facebook,instagram,whatsapp"}]

          

Reference Link:
   https://grokonez.com/spring-framework/spring-data/couchbase-create-spring-jpa-couchbase-application-springboot
   https://www.youtube.com/watch?v=VFEOgf-AexY&t=265s&ab_channel=CodeDebugger

### CODE REVIEW

**Review by jpukg — APPROVED** [review]

File: api/google-search/src/main/java/com/google/search/googlesearch/model/Search.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.google.search.googlesearch.model;
+
+import org.springframework.data.couchbase.core.mapping.id.GeneratedValue;
+import org.springframework.data.couchbase.core.mapping.id.GenerationStrategy;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.couchbase.core.mapping.Document;
+
+import lombok.Data;
```
Comment (ramsunvtech) [inline api/google-search/src/main/java/com/google/search/googlesearch/model/Search.java:8]: i guess this one missed in the documentation of PR #30

```
<dependency>
     <groupId>org.projectlombok</groupId>
    <artifactId>lombok</artifactId>
</dependency>
```

File: api/google-search/src/main/java/com/google/search/googlesearch/controller/v1/api/SearchController.java [hunk 2]
```diff
@@ -31,4 +44,14 @@
 
         return response;
     }
+
+    @PostMapping(produces = MediaType.APPLICATION_JSON_VALUE, value = "v1/addSearch")
+    public Search addSearchResult(@RequestBody Search search) {
+        return searchRepository.save(search); 
+    }
+
+    @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE, value = "v1/getSearch")
```
Comment (ramsunvtech) [inline api/google-search/src/main/java/com/google/search/googlesearch/controller/v1/api/SearchController.java:53]: @jpukg 
is value fine `v1/getSearch` ????

**Review by ramsunvtech — APPROVED** [review]

**Review by naveenvignesh5 — APPROVED** [review]

File: api/google-search/src/main/java/com/google/search/googlesearch/controller/v1/api/SearchController.java (hunk 2 above)
Comment (shangan23) [reply api/google-search/src/main/java/com/google/search/googlesearch/controller/v1/api/SearchController.java:53]: @karthikeyanpa12 @ramsunvtech  

Let's have the URL as `v1/search` for CRUD operations.

- For creating it's POST -> `v1/search`
- For reading it's GET -> `v1/search`
- For updating it's PUT -> `v1/search`
- For Deleted it's DELETE -> `v1/search`

**Review by shangan23 — CHANGES_REQUESTED** [review]
Request you to keep the URL consistent for a module. 
Let it be differentiated by HTTP VERBs

**Review by shangan23 — APPROVED** [review]
Approved

**Pull request merged**